apps/bfd-pipeline/bfd-pipeline-rda-grpc-server/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RdaServerApp.java

```diff
@@ -46,35 +50,27 @@ public static void main(String[] args) throws Exception {
     private final int maxToSend;
     @Nullable private final File fissClaimFile;
     @Nullable private final File mcsClaimFile;
+    @Nullable private final S3JsonMessageSources s3Sources;
 
     private Config(String[] args) throws Exception {
-      int port = 5003;
-      long seed = System.currentTimeMillis();
-      int maxToSend = 5_000;
-      File fissClaimFile = null;
-      File mcsClaimFile = null;
-      for (String arg : args) {
-        if (arg.startsWith("port:")) {
-          port = Integer.parseInt(arg.substring(5));
-        } else if (arg.startsWith("maxToSend:")) {
-          maxToSend = Integer.parseInt(arg.substring(10));
-        } else if (arg.startsWith("seed:")) {
-          seed = Long.parseLong(arg.substring(5));
-        } else if (arg.equals("random")) {
-          seed = Long.parseLong(arg.substring(5));
-        } else if (arg.startsWith("fissFile:")) {
-          fissClaimFile = new File(arg.substring(9));
-        } else if (arg.startsWith("mcsFile:")) {
-          mcsClaimFile = new File(arg.substring(8));
-        } else {
-          throw new IOException("invalid argument: " + arg);
-        }
+      final ConfigLoader config =
+          ConfigLoader.builder().addKeyValueCommandLineArguments(args).build();
+      port = config.intValue("port", 5003);
+      seed = config.longOption("seed").orElseGet(System::currentTimeMillis);
+      maxToSend = config.intValue("maxToSend", 5_000);
+      fissClaimFile = config.readableFileOption("fissFile").orElse(null);
+      mcsClaimFile = config.readableFileOption("mcsFile").orElse(null);
+      final Optional<String> s3Bucket = config.stringOption("s3Bucket");
```

Comment:
thoughts about using a path style option here rather than just the bucket name.  I'm thinking that with the work the BFD team did last sprint in adding minio support, we could test this out locally using a minio path running an a local docker network?

Reply:
Yes I have tested this using Minio locally.  A bucket maps to a directory inside the minio data directory.  It doesn't allow path characters in the bucket name.